src/listener/osx_keyboard_listener.rs

```diff
@@ -0,0 +1,40 @@
+use livesplit_hotkey::{Hook, Hotkey, KeyCode, Modifiers};
+use std::{error::Error, thread};
+
+use common::KeyboardListenerProvider;
+
+use super::common;
+pub struct OSXKeyboardListener {
+    hook: Hook,
+    hotkey: Hotkey,
+    callback: fn(),
+}
+
+impl KeyboardListenerProvider for OSXKeyboardListener {
+    fn new(callback: fn()) -> Result<OSXKeyboardListener, Box<dyn Error>> {
+        return Ok(OSXKeyboardListener {
+            hook: Hook::new().unwrap(),
+            hotkey: Hotkey {
+                key_code: KeyCode::KeyC,
+                modifiers: Modifiers::META,
+            },
+            callback,
+        });
+    }
+
+    fn start(&mut self) -> Result<&str, Box<dyn Error>> {
+        let cb = self.callback;
+        self.hook
+            .register(self.hotkey, move || {
+                thread::spawn(cb);
```

Comment:
this gives me hope in the Async environment. however, notice this can only work if the cb is empty and does not require "self", in our socket situation we must wait for an input but when we want the input-ready callback to be called we need the socket as well to read from (which makes this whole stuff very complicated)